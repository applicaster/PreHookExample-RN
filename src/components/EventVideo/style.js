import {
  
  Dimensions,
  StyleSheet,
} from 'react-native';

const screenWidth = Dimensions.get('window').width;
export const styles = StyleSheet.create({
  eventVideo: {
    position: 'absolute',
    top: 0,
    left: 0,
    zIndex: 1,
  },
  mediaItem: {
    height: screenWidth,
    width: screenWidth,
  },
});
