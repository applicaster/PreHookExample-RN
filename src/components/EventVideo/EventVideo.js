import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Dimensions,
  TouchableWithoutFeedback,
} from 'react-native';
import Video from 'react-native-video';
import { styles } from './style';

class EventVideo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      muted: true,
    };

    this.toggleAudio = this.toggleAudio.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    const { eventId } = this.props;
    const { eventIdForActiveAudio: nextEventIdForActiveAudio } = nextProps;

    if (eventId !== nextEventIdForActiveAudio) {
      this.setState({ muted: true });
    }
  }

  getMediaItemStyles() {
    const { width, height } = this.props;
    const screenWidth = Dimensions.get('window').width;
    let mediaItemStyles = styles.mediaItem;
    
    if (width !== height) {
      const aspectRatio = (width / height);
      mediaItemStyles = {
        height: (screenWidth / aspectRatio),
        width: screenWidth,
      };
    }

    return mediaItemStyles;
  }

  toggleAudio() {
    const { eventId, setEventIdForActiveAudio } = this.props;
    const { muted } = this.state;
    this.setState({ muted: !muted });
    setEventIdForActiveAudio(eventId);
  }

  render() {
    const { videoUrl } = this.props;
    if (!videoUrl) return null;
  
    const { muted } = this.state;
    return (<TouchableWithoutFeedback onPress={this.toggleAudio}>
      <Video
        source={{ uri: videoUrl }}
        muted={muted}
        resizeMode="cover"
        repeat
        style={[this.getMediaItemStyles(), styles.videoItem]}
      />
    </TouchableWithoutFeedback>);
  }
}

EventVideo.propTypes = {
  eventId: PropTypes.string,
  eventIdForActiveAudio: PropTypes.string,
  videoUrl: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
  setEventIdForActiveAudio: PropTypes.func,
};

export default EventVideo;
