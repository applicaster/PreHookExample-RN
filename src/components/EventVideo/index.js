import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setEventIdForActiveAudio } from '../../actions';
import { getEventIdForActiveAudio } from '../../selectors';

import EventVideo from './EventVideo';

const mapStateToProps = state => ({
  eventIdForActiveAudio: getEventIdForActiveAudio(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  setEventIdForActiveAudio,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(EventVideo);
