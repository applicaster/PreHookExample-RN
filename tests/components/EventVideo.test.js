import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import EventVideo from '../../src/components/EventVideo/EventVideo';

jest.mock('react-native-video', () => 'Video');

const eventVideoProps = (height, width, videoUrl) => ({
  height,
  width,
  videoUrl,
});

describe('EventVideo', () => {
  test('component renders correctly with no video url', () => {
    const html = renderer.create(
      <EventVideo />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly', () => {
    const html = renderer.create(
      <EventVideo {...eventVideoProps(100, 100, 'url')} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
