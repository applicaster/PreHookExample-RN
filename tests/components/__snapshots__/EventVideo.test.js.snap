// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventVideo component renders correctly 1`] = `
<Video
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  muted={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  repeat={true}
  resizeMode="cover"
  source={
    Object {
      "uri": "url",
    }
  }
  style={
    Array [
      Object {
        "height": 710,
        "width": 710,
      },
      undefined,
    ]
  }
  testID={undefined}
/>
`;

exports[`EventVideo component renders correctly with no video url 1`] = `null`;
